src/borrower/feed/navfeed.sol

```diff
@@ -3,267 +3,155 @@ pragma solidity >=0.6.12;
+}
+
+interface PileLike {
+    function setRate(uint loan, uint rate) external;
+    function debt(uint loan) external view returns (uint);
+    function pie(uint loan) external returns (uint);
+    function changeRate(uint loan, uint newRate) external;
+    function loanRates(uint loan) external view returns (uint);
+    function file(bytes32, uint, uint) external;
+    function rates(uint rate) external view returns (uint, uint, uint ,uint48, uint);
+    function total() external view returns (uint);
+    function rateDebt(uint rate) external view returns (uint);
+}
 
-// The Nav Feed contract extends the functionality of the NFT Feed by the Net Asset Value (NAV) computation of a Tinlake pool.
+// The NAV Feed contract extends the functionality of the NFT Feed by the Net Asset Value (NAV) computation of a Tinlake pool.
 // NAV is computed as the sum of all discounted future values (fv) of ongoing loans (debt > 0) in the pool.
 // The applied discountRate is dependant on the maturity data of the underlying collateral. The discount decreases with the maturity date approaching.
 // To optimize the NAV calculation the discounting of future values happens bucketwise. FVs from assets with the same maturity date are added to one bucket.
 // This safes iterations & gas, as the same discountRates can be applied per bucket.
-contract NAVFeed is BaseNFTFeed, Interest, Buckets, FixedPoint {
-
-    // maturityDate is the expected date of repayment for an asset
-    // nftID => maturityDate
-    mapping (bytes32 => uint) public maturityDate;
-
-    // recoveryRatePD is a combined rate that includes the probability of default for an asset of a certain risk group and its recovery rate
-    // risk => recoveryRatePD
-    mapping (uint => Fixed27) public recoveryRatePD;
-
-    // futureValue of an asset based on the loan debt, interest rate, maturity date and recoveryRatePD
-    // nftID => futureValue
-    mapping (bytes32 => uint) public futureValue;
-
-    WriteOff [2] public writeOffs;
-
-    struct WriteOff {
-        uint rateGroup;
+abstract contract NAVFeed is Auth, Discounting, DSTest {
+
+    PileLike    public pile;
+    ShelfLike   public shelf;
+
+    mapping (uint => uint)      public ceilingRatio;    // risk => ceilingRatio
+    mapping (bytes32 => uint)   public maturityDate;    // nftID => maturityDate
+    mapping (uint => uint)      public thresholdRatio;  // risk => thresholdRatio
+    mapping (uint => Fixed27)   public recoveryRatePD;  // risk => recoveryRatePD
+    mapping (bytes32 => uint)   public risk;            // nftID => risk
+    mapping (bytes32 => uint)   public nftValues;       // nftID => nftValues
+    mapping (bytes32 => uint)   public futureValue;     // nftID => futureValue
+    mapping (uint => uint)      public borrowed;        // loan => borrowed
+    mapping (uint => bool)      public writeOffOverride;// loan => writeOffOverride
+    mapping (uint => uint)      public buckets;         // timestamp => bucket
```

Comment:
I think we should look into optimisation here. 
If we use a struct, we can use uint128 and uint64 etc.

Would be nice to compare the gas differences etc.

Reply:
Would be nice, but I think we should look into this after we have all the functionality of NAV v2 working, including proper tests etc.